### PR TITLE
Update README URLs based on HTTP redirects

### PR DESCRIPTION
**Pull request by ReadmeCritic**
Created with https://github.com/dkhamsing/frankenstein

### GitHub Corrected URLs 
Was | Now 
--- | --- 
https://github.com/kazu-yamamoto/ghc-mod/issues/615 | https://github.com/DanielG/ghc-mod/issues/615 

### HTTPS Corrected URLs 
Was | Now 
--- | --- 
http://travis-ci.org/#alanz/HaRe | https://travis-ci.org/#alanz/HaRe 
http://www.behance.net/christidutoit | https://www.behance.net/christidutoit 

### Other Corrected URLs 
Was | Now 
--- | --- 
http://hspec.github.com/ | http://hspec.github.io/ 
https://rawgithub.com/alanz/HaRe/master/HaReLogo.svg | https://rawgit.com/alanz/HaRe/master/HaReLogo.svg

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -129,17 +129,17 @@ To run the test suite do:
...
 
 ### Resources
 
   * [GHC chapter](http://aosabook.org/en/ghc.html) of
     [AOSA](http://aosabook.org "Architecture of Open Source
     Applications") (if only for the diagram of GHC phases and data structures)
-  * [GHC 7.6.3 API docs](http://www.haskell.org/ghc/docs/7.6.3/html/libraries/ghc-7.6.3/GHC.html)
-  * [GHC 7.4.2 API docs](http://www.haskell.org/ghc/docs/7.4.2/html/libraries/ghc-7.4.2/GHC.html)
+  * [GHC 7.6.3 API docs](https://downloads.haskell.org/~ghc/7.6.3/docs/html/libraries/ghc-7.6.3/GHC.html)
+  * [GHC 7.4.2 API docs](https://downloads.haskell.org/~ghc/7.4.2/docs/html/libraries/ghc-7.4.2/GHC.html)
```
Comment (alanz) [inline README.md]: I am not sure that these two are appropriate

Comment (ReadmeCritic) [inline README.md]: Will revert

**Comment by ReadmeCritic** [comment]
@alanz ready

**Comment by alanz** [comment]
Thanks

**Comment by ReadmeCritic** [comment]
:tada:

**Pull request merged**